pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+additional high level settings that are feature specific. Each feature contains one or more components, which
+correspond to Istio components in the cluster.
+
+k8s API: this API is a pass through to k8s resource settings for Istio k8s resources. It allows customizing Istio k8s
+resources like Affinity, Resource requests/limits, PodDisruptionBudgetSpec, Selectors etc. in a more consistent and
+k8s specific way compared to values.yaml. See KubernetesResourcesSpec in this file for details.
+
+2. values.yaml
+
+The entirety of values.yaml settings is accessible through InstallSpec. They are included for backward compatibility,
+but also because many of the values will be moved to configuration CRs that are consumed directly by Istio components
+rather than being install time parameters. The only remaining fields will be k8s resource settings which will be
+replaced by the new API above.
+
+Values.yaml profiles are also embedded in the operator, wrapped inside InstallSpecs CommonComponentSpec/ValueOverrides
+field.
+
+3. k8s resource overlays
+
+Once a manifest is rendered from InstallSpec, a further specialization can be applied by specifying k8s resource
+overlays. The concept is similar to kustomize, where JSON patches are applied for object paths. This allows
+customization at the lowest level and eliminates the need to create ad-hoc template parameters.
+
+EXAMPLES
+
+1. Default Istio install
+
+spec: ~
+
+2. Default remote profile install
+
+spec:
+  baseSpecPath: istio-remote.yaml
+
+3. Default install with telemetry disabled
+
+spec:
+  telemetry:
+    enabled: false
+
+4. Default install with each feature installed to different namespace and security components in separate namespaces
+
+spec:
+  traffic_management:
+    namespace: istio-traffic-management
+  policy:
+    namespace: istio-policy
+  telemetry:
+    namespace: istio-telemetry
+  config_management:
+    namespace: istio-config-management
+  security:
+    citadel:
+      namespace: istio-citadel
+    cert_manager:
+      namespace: istio-cert-manager
+    node_agent:
+      namespace: istio-node-agent
+
+5. Default install with specialized k8s settings for pilot
+
+spec:
+  traffic_management:
+    pilot:
+        k8s:
+          resources:
+            limits:
+              cpu: 444m
+              memory: 333Mi
+            requests:
+              cpu: 222m
+              memory: 111Mi
+          readinessProbe:
+            failureThreshold: 44
+            initialDelaySeconds: 11
+            periodSeconds: 22
+            successThreshold: 33
+
+6. Default install with values.yaml customizations for proxy
+
+spec:
+  traffic_management:
+    proxy:
+      valuesOverrides:
+      - global.proxy.enableCoreDump: true
+      - global.proxy.dnsRefreshRate: 10s
+
+7. Default install with modification to container flag in galley
+
+spec:
+  configuration_management:
+    galley:
+      k8s:
+        overlays:
```

Comment:
why overlays here?

Reply:
is it for overwrite?

Reply:
so is the intention here you can use valuesOverrides to override values that can be configured via helm value, then use yaml style overrides to overwrite yaml blobs?

Reply:
yes, exactly.